gui/create-item.lua

```diff
@@ -184,11 +211,176 @@ local function getCreatureRaceAndCaste(caste)
+ local corpseTable = {BONE = "BONE", SKIN = "SKIN", CARTILAGE = "CARTILAGE", TOOTH = "TOOTH", NERVE = "NERVE", NAIL = "NAIL", HORN = "HORN", HOOF = "HOOF"}
+ -- we do the same as above but with hair
+ local hairTable = {HAIR = "HAIR", EYEBROW = "EYEBROW", EYELASH = "EYELASH", MOUSTACHE = "MOUSTACHE", CHIN_WHISKERS = "CHIN_WHISKERS", SIDEBURNS = "SIDEBURNS"}
+ -- if the layer is fat, spawn a glob of fat and DON'T check for other layer types
+ if layerName == "FAT" then
+  item_type = "GLOB"
+ elseif corpseTable[layerName] or hairTable[layerName] then -- check if hair
+  -- farlayer = 5
+  -- layerName = "HAIR" -- we then simplify every hair-tissue into just "HAIR"
+  item_type = "CORPSEPIECE"
+ end
+ 
+ local itemType = dfhack.items.findType(item_type..":NONE")
+ local itemSubtype = dfhack.items.findSubtype(item_type..":NONE")
+ local material = "CREATURE_MAT:"..raceName..":"..layerMat
+ local materialInfo = dfhack.matinfo.find(material)
+ 
+ local item_id = dfhack.items.createItem(itemType, itemSubtype, materialInfo['type'], materialInfo.index, creator)
+ local item = df.item.find(item_id)
+ 
+ item:setQuality(quality)
+ 
+ local isCorpsePiece = false
+ 
+ -- if the item type is a corpsepiece, we know we have one, and then go on to set the appropriate flags
+ if item_type == "CORPSEPIECE" then
+	 if layerName == "BONE" then -- check if bones
+		item.corpse_flags.bone = true
+		item.material_amount.Bone = 1
+	 elseif layerName == "SKIN" then -- check if skin/leather
+		item.corpse_flags.leather = true
+		item.material_amount.Leather = 1
+	 -- elseif layerName == "CARTILAGE" then -- check if cartilage (NO SPECIAL FLAGS)
+	 elseif layerName == "HAIR" then -- check if hair (simplified from before)
+		item.corpse_flags.hair_wool = true
+		item.material_amount.HairWool = 1
+		if materialInfo.material.flags.YARN then
+			item.corpse_flags.yarn = true
+			item.material_amount.Yarn = 1
+		end
+	 elseif layerName == "TOOTH" then -- check if tooth
+		item.corpse_flags.tooth = true
+		item.material_amount.Tooth = 1
+	 elseif layerName == "NERVE" then -- check if nervous tissue
+		item.corpse_flags.skull1 = true -- apparently "skull1" is supposed to be named "rots/can_rot"
+		item.corpse_flags.separated_part = true
+	 -- elseif layerName == "NAIL" then -- check if nail (NO SPECIAL FLAGS)
+	 elseif layerName == "HORN" or layerName == "HOOF" then -- check if nail
+		item.corpse_flags.horn = true
+		item.material_amount.Horn = 1
+		isCorpsePiece = true
+	 end
+	 -- checking for skull
+	 if not generic and creatorBody.body_parts[bodypart].token == "SKULL" then
+	  item.corpse_flags.skull2 = true
+	 end
+ end
+ 
+ local matType
+ 
+ -- figure out which material type the material is (probably a better way of doing this but whatever)
+ for i in pairs(creatorRaceRaw.tissue) do
+  if creatorRaceRaw.tissue[i].tissue_material_str[1] == layerMat then
+	matType = creatorRaceRaw.tissue[i].mat_type
```

Comment:
is there a chance that this is not found?

Reply:
wouldn't that mean that a tissue just,        doesn't have a material?

Reply:
did the other things but still unsure on this. _can_ this happen?